docs/planning/sprintplanning.md

```diff
@@ -0,0 +1,66 @@
+- priority/p0 is reserved for critical bugs that needs to be addressed immediately in current release at the cost of currently planned items.
+- WG lead will drive this and his/her decision will be final
+
+## How to manage Story/Feature?
+- Story/Feature typically spans 1-2 weeks for one engineer.
+To propose a new feature,  create a new issue of type “Feature Request”
+- Please fill in as much detail as possible in the Feature template and propose a release cycle
+- All new features will be labelled “kind/feature”, and “triage/needstriage”
+- All new proposals are discussed either in WG meeting, or on slack, or dedicated meetings if needed at any point scheduled by the proposer. 
+- By R-2w’s WG meeting all new proposals should be vetted and verified that they fit the [Planning criteria](#planning-criteria). Features that do not fit this criteria will not be considered in the current release planning cycle and hence early discussion is encouraged instead of waiting for R-2w.
+- All eligible proposals will then be snapshotted into the voting document and open for voting till R-1w WG meeting. In R-1w WG meeting top features based on votes will be planned for next release. 
+> We need some kind of estimates for each feature for release planning. Shall we adopt story points on fibonacci scale or t-shirt sizing? My vote goes to story points on fibonacci scale to avoid spending a lot of time on estimation, which is a common pitfall.
+- Selected features will be labeled “triage/accepted”. Rest will be labelled “triage/ice-boxed” and will be eligible for next planning cycle. Accepted features will be added to the correct milestone and sprint specific project for tracking.
+
+## How to manage Epic/Scenario?
+- Epic/Scenario typically spans 1-3 months or even more in some cases.
+- For new proposals these will be created and triaged/refined just like Story/Feature. However, after triaging these need to be broken down into Features for planning and voting.
+- Github doesn’t allow nested Issues, hence the only good way to track these are by creating Projects and adding the broken down features as issues to this project. An issue can be part of multiple projects and hence we can get a Scenario\Epic view as well as a Sprint view through respective projects.
+
+## Planning criteria:
+- Features are well defined and scoped with clear exit criteria. No feature creeps once release starts
+- Features can be implemented in <2w by one engineer. In case of Epic/Scenario, it could be broken down into <2w features with clear exit criteria.
+- The engineering team should be confident that any engineer can implement the feature based on the content in the feature *(This is similar to refinement in a scrum, but doesn’t have to be as strict as a functional specification)*.
+
+## Organizing with Github Projects 
+- New Project per sprint. Today, the sprint and milestone are both aligned. However in future we may decide to have multiple sprints in a release cycle.
```

Comment:
I'm not sure about this. In my mind sprints are "collections of work items predicted to be completed in a time period". That doesn't seem like a project to me because there's no collective goal. Practically, I think the short lifecycle and timed nature of these projects will lead to them feeling more like busy-work, and thus poorly organized and abandoned after the sprint deadline. These abandoned sprint projects will them obscure the story and scenario projects that are working toward a specific goal.

> in future we may decide to have multiple sprints in a release cycle.

Would we ever have sprints that are not aligned with the release cycle, e.g. the release deadline is 1 week into a 2-week sprint? To me sprints seem closer to releases and/or milestones than projects.

We will get this question a lot: "when do I use milestones and when do I use projects?" A simple, clear answer might be, "Use milestones when you have a deadline, otherwise projects." If we use projects for sprints, this answer get more complicated.

I'm not a project manager by nature so I could easily be wrong about all this!

Reply:
I agree with the concepts you explained here. However I found GitHub needs to do lot more work on providing the right artifacts for project management. Here is what I found:
 - Issue is being overloaded and extended into anything that can be tracked - So we use Issue to represent Bug, Feature, Epic etc with a label on it.
 - Project is the only way to get a board such as a kanban or a scrum board to get a summary of the state of the project. In reality these are two different constructs. But since there is no other way to create a Kanban board, the proposal was to create a project so that we get a board.
- Milestone is exactly what the name says. In our case it fits the use case of Release Date.

About sprints v/s releases. IMO, sprint should align with release milestone but are de-coupled. A sprint is also a milestone (internal) that the team has mutually agreed. However just for the sake of getting a board view of a sprint, we need a Github project. Unless there is other way :)
However, it wont be a problem now as we are planning 6 week's work which aligns with the release. In case we feel that we want to be more aggressive and predictable with planning and execution then we can make shorter sprints which eventually align with the release date, but are decoupled.

Do you think board will help? Or is there a way to get a board for a milestone?

Reply:
IMO shorter sprints are better, and we should aim to have three 2-week sprints or two 3-week ones. In my experience Parkinson's law (i.e., work expands so as to fill the time available for its completion) *always* occurs. If engineers have roughly three 2-week-long features assigned, I bet that *all* of them will be completed by the end of the 6 weeks, there will be a merging chaos closer to the actual release date, and no time for integration and system testing. Tracking progress and managing dependencies between features will also be harder.

Having said that I'm OK with starting with a pilot of a 6-week sprint if everyone agrees. But I think, at least, it would be better to plan for a 5-week sprint, and leave a 1-week reservation period for system testing, dog fooding, release related tasks, and any other problem that may arise.

Reply:
I think if we use [ZenHub](https://www.zenhub.com/) tracking will be very easy. I added a comment to the issue at the end. I'm not sure if we can use it though.